packages/core/src/deployment.ts

```diff
@@ -217,3 +281,38 @@ export class InvalidDeployment extends Error {
     return msg;
   }
 }
+
+/**
+ * Checks if the deployment id is completed.
+ *
+ * @param address The expected address of the deployment.
+ * @param deploymentId The deployment id.
+ * @param getDeploymentResponse A function to get the deployment response.
+ * @returns true if the deployment id is completed, false otherwise.
+ * @throws {InvalidDeployment} if the deployment id failed.
+ */
+export async function isDeploymentCompleted(
+  address: string,
+  deploymentId: string,
+  getDeploymentResponse: (allowUndefined: boolean) => Promise<DeploymentResponse | undefined>,
+): Promise<boolean> {
+  debug('verifying deployment id', deploymentId);
+  const response = await getDeploymentResponse(false);
+  if (response === undefined) {
+    throw new Error(`Broken invariant: Response not found for deployment id ${deploymentId}`);
+  }
```

Comment:
It seems like the only place where we use `allowUndefined = false` we check the result anyway. I would simplify `getDeploymentResponse` removing the `allowUndefined` argument, making it always return `undefined` if the deployment is not found.

In this function I would also make `getDeploymentResponse` a function that takes a deployment id, instead of hardcoding it at the callsite.

I think both of these changes should make invocation simpler because you would just pass along the `getDeploymentResponse` function.